# Importing libraries

In [1]:
import csv 
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np
from sklearn.datasets import load_digits
import matplotlib.pyplot as plt
digits = load_digits()


# Splitting the Data

In [2]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(digits.data,digits.target,test_size=0.3)

# Logistic Regreession

In [3]:
lr = LogisticRegression(solver='liblinear',multi_class='ovr')
lr.fit(X_train, y_train)
lr.score(X_test, y_test)

0.9537037037037037

# SVM 

In [4]:
svm = SVC(gamma='auto')
svm.fit(X_train, y_train)
svm.score(X_test, y_test)

0.5740740740740741

# Random Classifier

In [5]:
rf = RandomForestClassifier(n_estimators=40)
rf.fit(X_train, y_train)
rf.score(X_test, y_test)

0.9703703703703703

# K Fold cross validation


In [6]:
from sklearn.model_selection import KFold
kf = KFold(n_splits=3)
kf

KFold(n_splits=3, random_state=None, shuffle=False)

## Cross validating


In [7]:
from sklearn.model_selection import cross_val_score

In [8]:
cross_val_score(LogisticRegression(solver='liblinear',multi_class='ovr'), digits.data, digits.target,cv=3)


array([0.89534884, 0.94991653, 0.90939597])

In [9]:
cross_val_score(SVC(gamma='auto'), digits.data, digits.target,cv=3)


array([0.39368771, 0.41068447, 0.45973154])

In [10]:
cross_val_score(RandomForestClassifier(n_estimators=40),digits.data, digits.target,cv=3)


array([0.93355482, 0.95492487, 0.92281879])

In [11]:
#AVERAGE SCORE OF LINEAR REGRESSION
scores_lg = cross_val_score(LogisticRegression(solver='liblinear',multi_class='ovr'), digits.data, digits.target,cv=3)
np.average(scores_lg)

0.9182204459698582

In [12]:
#AVERAGE SCORE OF SVC
scores_svc = cross_val_score(SVC(gamma='auto'), digits.data, digits.target,cv=3)
np.average(scores_svc)

0.4213679084629655

In [13]:
#AVERAGE SCORE OF RANDOM FOREST
scores_rf = cross_val_score(RandomForestClassifier(n_estimators=40),digits.data, digits.target,cv=3)
np.average(scores_rf)

0.9382291555887162

# Result

By comparing the results</br>

### LINEAR REGRESSION - 91.82 %</br>

### SVM - 42.13 %</br>

## RANDOM FOREST - 93.82%</br>

Clearly Random Forest has the upper hand.🌳🌴🌲